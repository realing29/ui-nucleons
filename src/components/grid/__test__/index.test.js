import React from 'react';
import { Layout, Row, Column } from '../';
import GridEntity from '../grid-entity';
import GridLayout from '../grid-layout';
import GridRow from '../grid-row';
import GridCol from '../grid-col';
import { mount } from 'enzyme/build';

describe('GridLayout, GridRow, GridCol', () => {
  it('renders correctly', () => {
    const grid = mount(
      <Layout>
        <Row>
          <Column desktop={3}>column 1</Column>
          <Column desktop={5}>column 2</Column>
        </Row>
        <Row>
          <Column desktop={5}>column 3</Column>
        </Row>
      </Layout>
    );
    expect(grid.find(GridEntity)).toHaveLength(5);
    expect(grid.find(GridLayout)).toHaveLength(1);
    expect(grid.find(GridRow)).toHaveLength(2);
    expect(grid.find(GridCol)).toHaveLength(3);
    expect(grid).toMatchSnapshot();
  });
});
