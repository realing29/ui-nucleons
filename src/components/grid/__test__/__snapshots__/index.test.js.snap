// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridLayout, GridRow, GridCol renders correctly 1`] = `
<GridLayout>
  <div
    className="layout-wrapper layout-xxl-padding-sm layout-xl-padding-sm layout-lg-padding-sm layout-md-padding-lg layout-sm-padding-sm layout-xs-padding-sm layout-xxs-padding-md layout-xxxs-padding-sm"
  >
    <div
      className="layout layout-xxl-width-xl layout-xl-width-lg layout-lg-width-md layout-md-width-md layout-sm-width-sm layout-xs-width-xs layout-xxs-width-xs layout-xxxs-width-xs"
    >
      <GridRow>
        <GridEntity
          createClasses={[Function]}
        >
          <div
            className="row items-stretch justify-start with-gutters row-lg-columns-count-12 row-md-columns-count-8 row-lg-columns-gutters-lg row-md-columns-gutters-md row-sm-columns-gutters-sm"
          >
            <GridCol
              desktop={3}
            >
              <GridEntity
                createClasses={[Function]}
                desktop={3}
              >
                <div
                  className="col col-lg-width-3 col-md-width-auto"
                >
                  column 1
                </div>
              </GridEntity>
            </GridCol>
            <GridCol
              desktop={5}
            >
              <GridEntity
                createClasses={[Function]}
                desktop={5}
              >
                <div
                  className="col col-lg-width-5 col-md-width-auto"
                >
                  column 2
                </div>
              </GridEntity>
            </GridCol>
          </div>
        </GridEntity>
      </GridRow>
      <GridRow>
        <GridEntity
          createClasses={[Function]}
        >
          <div
            className="row items-stretch justify-start with-gutters row-lg-columns-count-12 row-md-columns-count-8 row-lg-columns-gutters-lg row-md-columns-gutters-md row-sm-columns-gutters-sm"
          >
            <GridCol
              desktop={5}
            >
              <GridEntity
                createClasses={[Function]}
                desktop={5}
              >
                <div
                  className="col col-lg-width-5 col-md-width-auto"
                >
                  column 3
                </div>
              </GridEntity>
            </GridCol>
          </div>
        </GridEntity>
      </GridRow>
    </div>
  </div>
</GridLayout>
`;
